src/app/static/src/tests/components/runWorkflow/runWorkflowReport/runWorkflowReport_validation.test.ts

```diff
@@ -0,0 +1,336 @@
+            wrapper.setData({selectedReport: "minimal"});
+            await Vue.nextTick();
+            wrapper.find("#add-report-button").trigger("click");
+
+            setTimeout(async () => {
+                updateComponentMetadataFromLastEmitted(wrapper);
+                await Vue.nextTick();
+                expect(wrapper.emitted("valid").length).toBe(2);
+                expect(wrapper.emitted("valid")[1][0]).toBe(false);
+                done();
+            });
+        });
+    });
+
+    it("emits valid true when workflow becomes valid after set parameter value", (done) => {
+        const wrapper = getWrapper({
+            reports: [{name: "global", params: {p1: null}}]
+        });
+        setTimeout(async () => {
+            expect(wrapper.emitted("valid")).toBeUndefined();
+
+            wrapper.find("#param-control-0").setValue("1");
+            await Vue.nextTick();
+            expectEmittedValid(wrapper);
+            done();
+        });
+    });
+
+    it("emits valid false when workflow becomes invalid after unset parameter value", (done) => {
+        const wrapper = getWrapper({
+            reports: [{name: "global", params: {p1: "v1"}}]
+        });
+        setTimeout(async () => {
+            expectEmittedValid(wrapper);
+
+            wrapper.find("#param-control-0").setValue("");
+            await Vue.nextTick();
+            expect(wrapper.emitted("valid").length).toBe(2);
+            expect(wrapper.emitted("valid")[1][0]).toBe(false);
+            done();
+        });
+    });
+
+    it("emits valid false when workflow becomes invalid when remove only report", (done) => {
+        const wrapper = getWrapper({
+            reports: [{name: "global"}]
+        });
+        setTimeout(async () => {
+            expectEmittedValid(wrapper);
+
+            wrapper.find(".remove-report-button").trigger("click");
+            await Vue.nextTick();
+            expect(wrapper.emitted("valid").length).toBe(2);
+            expect(wrapper.emitted("valid")[1][0]).toBe(false);
+            done();
+        });
+    });
+
+    it("emits valid true when workflow becomes valid when remove invalid report", (done) => {
+        const wrapper = getWrapper({
+            reports: [
+                {name: "global"},
+                {name: "minimal", params: {p1: null}}
+            ]
+        });
+        setTimeout(async () => {
+            expect(wrapper.emitted("valid")).toBeUndefined();
+
+            wrapper.findAll(".remove-report-button").at(1).trigger("click");
+
+            await Vue.nextTick();
+            expectEmittedValid(wrapper);
+            done();
+        });
+    });
+
+    it("emits valid true when commit change removes report which was invalid", (done) => {
+        mockAxios.onGet('http://app/reports/runnable/?branch=master&commit=abc123')
+            .reply(200, {"data": [{ name: "minimal", date: null }]});
+
+        mockAxios.onGet('http://app/report/minimal/parameters/?commit=abc123')
```

Comment:
Yeah, really good question, well spotted! I think the issue here is that if the component gets an error response when it checks the parameters in the new branch, it just shrugs its shoulders, assumes the parameters are fine and powers on with any remaining checks. It will display the error, but as far as it's concerned, the workflow is valid. 

I'll add some additional checking in the test that the error hasn't been set when we're not expecting it. 

Not sure if the component behaviour should change. I can't really think of a better alternative that wouldn't introduce even more complexity e.g. we could consider the report invalid due to error response, but would have to allow user some way of attempting to re-validate. Definitely one for another ticket though!